sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -68,8 +69,14 @@ def _update_headers(headers, token):
     @property
     def _need_new_token(self):
         # type: () -> bool
-        return not self._token or self._token.expires_on - time.time() < 300
-
+        return not self._token or self._token.expires_on - time.time() < self._token_refresh_offset
+
+    def _load_token_refresh_options(self):
+        try:
+            token_refresh_options = self._credential.get_token_refresh_options()
+            self._token_refresh_offset = int(token_refresh_options.get("token_refresh_offset", 300))
+        except Exception:  # pylint: disable=broad-except
```

Comment:
I would just catch AttributeError (if get_token_refresh_options is not defined), and let a potential "NotANumber" throw from the int conversion